system-helper/flatpak-system-helper.c

```diff
@@ -141,6 +206,137 @@ no_progress_cb (OstreeAsyncProgress *progress, gpointer user_data)
 {
 }
 
+static uid_t
+get_connection_uid (GDBusMethodInvocation *invocation, GError **error)
+{
+  GDBusConnection *connection = g_dbus_method_invocation_get_connection (invocation);
+  const gchar *sender = g_dbus_method_invocation_get_sender (invocation);
+  g_autoptr(GVariant) credentials = NULL;
+  g_autoptr(GVariantIter) iter = NULL;
+  const char *key = NULL;
+  g_autoptr(GVariant) value = NULL;
+  uid_t uid;
+
+  credentials = g_dbus_connection_call_sync (connection,
+                                             "org.freedesktop.DBus",
+                                             "/org/freedesktop/DBus",
+                                             "org.freedesktop.DBus",
+                                             "GetConnectionCredentials",
+                                             g_variant_new ("(s)", sender),
+                                             NULL, G_DBUS_CALL_FLAGS_NONE, G_MAXINT, NULL, error);
```

Comment:
Pass `G_VARIANT_TYPE ("(a{sv})")` for the (currently-`NULL`) `reply_type` parameter here.